lldb/source/Plugins/ObjectFile/XCOFF/ObjectFileXCOFF.cpp

```diff
@@ -136,13 +169,42 @@ bool ObjectFileXCOFF::MagicBytesMatch(DataBufferSP &data_sp,
   return XCOFFHeaderSizeFromMagic(magic) != 0;
 }
 
-bool ObjectFileXCOFF::ParseHeader() { return false; }
+bool ObjectFileXCOFF::ParseHeader() {
+
+  bool retVal = false;
+  ModuleSP module_sp(GetModule());
+  if (module_sp) {
+    std::lock_guard<std::recursive_mutex> guard(module_sp->GetMutex());
```

Comment:
Given that this is called directly from CreateInstance, I'm pretty sure this locking is not needed (ObjectFileELF definitely doesn't do that, although PECOFF for some reason does).

Reply:
Done